### ISSUE TITLE
Rule proposal: No default value in setter method

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#2577]
### Description

```js
class X {
	set text(value = '') {
		this.#text = value;
	}
}
```

When setting `undefined` to the property, the `''` default is not used here.

Relevant pull request: https://github.com/sindresorhus/yocto-spinner/pull/5

### Examples

```js
// ❌
set text(value = '') {
	this.#text = value;
}

// ✅
set text(value) {
	this.#text = value ?? '';
}
```

### Proposed rule name

no-default-setter-value

**Comment by fisker** [comment]
Are you sure about this?

```
(new class X {
	set text(value = '') {
		console.log({value})
	}
}).text = undefined

VM372:3 {value: ''}
```

**Comment by sindresorhus** [comment]
Hmm. Lesson learned. Should not trust random PRs. I got fooled into thinking it was correct since a saw a screenshot of the TS error. But I guess the limitation is only with TS.

**Comment by fisker** [comment]
> Default function parameters allow named parameters to be initialized with default values if no value or undefined is passed.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters

**Comment by sindresorhus** [comment]
> Maybe this can still useful?

That's not specific to setters, but rather how default parameters work. We have `no-null` for this already.
